src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -112,19 +112,15 @@ public static KeyToValueMappingEstimator MapKeyToValue(this TransformsCatalog.Co
            => new ValueToKeyMappingEstimator(CatalogUtils.GetEnvironment(catalog), inputColumn, outputColumn, maxNumTerms, sort);
 
         /// <summary>
-        /// Converts value types into <see cref="KeyType"/> loading the keys to use from <paramref name="file"/>.
+        /// Converts value types into <see cref="KeyType"/>, optionally loading the keys to use from <paramref name="termData"/>.
         /// </summary>
         /// <param name="catalog">The categorical transform's catalog.</param>
         /// <param name="columns">The data columns to map to keys.</param>
-        /// <param name="file">The path of the file containing the terms.</param>
-        /// <param name="termsColumn"></param>
-        /// <param name="loaderFactory"></param>
+        /// <param name="termData">The data view containing the terms. If unspecified, they will be
```

Comment:
Is it worth to specify what that dataview should have only one column and content of that column would be used to to specify allowed keys?
Can we also call it keyData?
I don't think we define Terms anywhere other than in maxNumTerms and even there in description we fallback to term keys.

Reply:
Good idea. I also did not like the name, but didn't quite know exactly to call it. Fixed this here and I hope everywhere, and also made another few opportunistic name changes that I realized no longer applied. Also enhanced the documentation as requested.